### ISSUE TITLE
[Bug]: When streaming chunks openrouter -> AttributeError: 'NoneType' object has no attribute 'content' 

### ISSUE THREAD

**Issue by EsoCoding** [BerriAI/litellm#1659]
### What happened?

With the latest versions i got a AttributeError: 'NoneType' object has no attribute 'content when streaming response from openrouter models. I did not test wether it also did this with others providers, i have updated the litellm to the latest version i believe and it already happend a while, but it caused a discconnect on the websocket, so i found a fix that worked for openrouter in the end.

```

                    os.environ["OPENROUTER_API_KEY"] = os.getenv("OPENROUTER_API_KEY")
                    
                    try:
                        response = completion(
                            model= self.selected_model,
                            messages=trim_messages(self.sessions[session_id], self.selected_model, trim_ratio=0.75, return_response_tokens=True, max_tokens=self.chart_data['chart-max-output']),
                            stream=True,
                            temperature=float(self.chart_data['chart-temp']),
                            max_tokens=int(self.chart_data['chart-max-output']),
                            top_p=float(self.chart_data['chart-topp']),
                            timeout=2,
                        )

                        for part in response:
                            
                            # Send the response chunk; include the markup_report if this is the last chunk
                            await websocket.send_json({
                                "message": part['choices'][0]['delta']['content'],
                                "report": markup_report,
                                "role": "assistant"
                            })
                    except APITimeoutError as e:
                        print(f"error occurred: {e} - {traceback.format_exc()}")
                        await websocket.send_json(
                            {"message": "Error message: {e}", "report": markup_report, "role": "assistant",}
                        )
                    except Exception as e:
                        print(f"error occurred: {e} - {traceback.format_exc()}")
                        await websocket.send_json(
                            {"message": "Error message: {e}", "report": markup_report, "role": "assistant",}
                        )
```JHILJK

### Relevant log output

```shell
error occurred: 'NoneType' object has no attribute 'content' - Traceback (most recent call last):
  File "/home/esocode/.cache/pypoetry/virtualenvs/sidtime-5_XSMlGe-py3.11/lib/python3.11/site-packages/litellm/utils.py", line 5987, in exception_type
    request=original_exception.request,
            ^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'AttributeError' object has no attribute 'request'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/esocode/Projects/sidtime/sidtime/app.py", line 145, in chat_socket
    for part in response:
  File "/home/esocode/.cache/pypoetry/virtualenvs/sidtime-5_XSMlGe-py3.11/lib/python3.11/site-packages/litellm/utils.py", line 8151, in __next__
    raise e
  File "/home/esocode/.cache/pypoetry/virtualenvs/sidtime-5_XSMlGe-py3.11/lib/python3.11/site-packages/litellm/utils.py", line 8132, in __next__
    response = self.chunk_creator(chunk=chunk)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/esocode/.cache/pypoetry/virtualenvs/sidtime-5_XSMlGe-py3.11/lib/python3.11/site-packages/litellm/utils.py", line 8093, in chunk_creator
    raise exception_type(
          ^^^^^^^^^^^^^^^
  File "/home/esocode/.cache/pypoetry/virtualenvs/sidtime-5_XSMlGe-py3.11/lib/python3.11/site-packages/litellm/utils.py", line 7008, in exception_type
    raise original_exception
  File "/home/esocode/.cache/pypoetry/virtualenvs/sidtime-5_XSMlGe-py3.11/lib/python3.11/site-packages/litellm/utils.py", line 7987, in chunk_creator
    response_obj = self.handle_openai_chat_completion_chunk(chunk)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/esocode/.cache/pypoetry/virtualenvs/sidtime-5_XSMlGe-py3.11/lib/python3.11/site-packages/litellm/utils.py", line 7546, in handle_openai_chat_completion_chunk
    raise e
  File "/home/esocode/.cache/pypoetry/virtualenvs/sidtime-5_XSMlGe-py3.11/lib/python3.11/site-packages/litellm/utils.py", line 7513, in handle_openai_chat_completion_chunk
    if str_line.choices[0].delta.content is not None:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'content'
```

**Comment by ishaan-jaff** [comment]
Trying to repro this

**Comment by ishaan-jaff** [comment]
Unable to repro it, this is my code snippet:
```python
response = litellm.completion(
            model="openrouter/mistralai/mistral-tiny",
            messages=messages,
            stream=True,
        )
for chunk in response:
    print(chunk)
```

**Comment by ishaan-jaff** [comment]
@EsoCoding what model were you using?

**Comment by ghost** [comment]
fix in prod (should be live in v`1.20.9` onwards

**Comment by EsoCoding** [comment]
> @EsoCoding what model were you using?

It happend with all models. I finally fixed the code myself, but it might break other providers so i didnt find itn suitable to send as a fix.

**Comment by tattwamasi** [comment]
I am seeing this now too with gpt-3.5-turbo in my initial, naive 1st try at just dropping in LiteLLM to replace my working app's OpenAI acompletion call.  
`DEBUG:    > TEXT '42["stream_token",{"id":"c7911108-5c8e-4817-aad...o","isSequence":false}]' [99 bytes]
2024-02-07 13:24:28 - 'Delta' object has no attribute 'role'
Traceback (most recent call last):
  File "[snip]/anaconda3/lib/python3.11/site-packages/pydantic/main.py", line 759, in __getattr__
    return pydantic_extra[item]
           ~~~~~~~~~~~~~~^^^^^^
KeyError: 'role'
`
I'm also getting an unexpected exception when I try to use my original default model of gpt-4-turbo-preview, so I recognize I may be missing something basic or have a conflict with chainlit (the UI kit I'm using) instrumentation or something.  For reference, the gpt4 error is:
`
Logging Details LiteLLM-Failure Call
Exception in route_message acompletion: LLM Provider NOT provided. Pass in the LLM provider you are trying to call. You passed model=gpt-4-turbo-preview
 Pass model as E.g. For 'Huggingface' inference endpoints pass in `completion(model='huggingface/starcoder',..)` Learn more: https://docs.litellm.ai/docs/providers<built-in method with_traceback of BadRequestError object at 0x127c2c100>
2024-02-07 13:17:08 - None`

**Comment by ghost** [comment]
@EsoCoding did the fix pushed from v`1.20.9` not work?
